docs/reference/builder.md

```diff
@@ -106,27 +106,175 @@ repository to its registry*](../userguide/containers/dockerrepos.md#contributing
 Here is the format of the `Dockerfile`:
 
-    # Comment
-    INSTRUCTION arguments
+```Dockerfile
+# Comment
+INSTRUCTION arguments
+```
+
+The instruction is not case-sensitive. However, convention is for them to
+be UPPERCASE to distinguish them from arguments more easily.
+
+
+Docker runs instructions in a `Dockerfile` in order. **The first 
+instruction must be \`FROM\`** in order to specify the [*Base
+Image*](glossary.md#base-image) from which you are building. 
+
+Docker treats lines that *begin* with `#` as a comment, unless the line is 
+a valid [parser directive](builder.md#parser directives). A `#` marker anywhere
+else in a line is treated as an argument. This allows statements like:
+
+```Dockerfile
+# Comment
+RUN echo 'we are running some # of cool things'
+```
+
+Line continuation characters are not supported in comments or parser directives.
+
+## Parser directives
 
-The instruction is not case-sensitive, however convention is for them to
-be UPPERCASE in order to distinguish them from arguments more easily.
+Parser directives are optional, and affect the way in which subsequent lines 
+in a `Dockerfile` are handled. Parser directives do not add layers to the build,
+and will not be shown as a build step. Parser directives must appear 
+before any instruction in a Dockerfile, and may be included once, and once only,
+in a Dockerfile. Parser directives are written as a special type of comment in
+the form `# DIRECTIVE=value`.  Due to these rules, the following examples are all 
+invalid:
 
-Docker runs the instructions in a `Dockerfile` in order. **The
-first instruction must be \`FROM\`** in order to specify the [*Base
-Image*](glossary.md#base-image) from which you are building.
+Invalid due to line continuation:
 
-Docker will treat lines that *begin* with `#` as a
-comment. A `#` marker anywhere else in the line will
-be treated as an argument. This allows statements like:
+```Dockerfile
+# DIREC \
+TIVE=value
+```
+
+Invalid due to appearing twice:
 
-    # Comment
-    RUN echo 'we are running some # of cool things'
+```Dockerfile
+# DIRECTIVE=value1
+FROM ImageName
+# DIRECTIVE=value2
+```
+    
+Invalid due to appearing after a builder instruction:
 
-Here is the set of instructions you can use in a `Dockerfile` for building
-images.
+```Dockerfile
+FROM ImageName
+# DIRECTIVE=value
+```
+
+The following would be treated as a comment due to the directive not being
+recognized:
+
+```Dockerfile
+# UNRECOGNIZEDDIRECTIVE=value
+```    
+    
+Directives are not case-sensitive.  However, convention is for them to
+be UPPERCASE to distinguish them from other comments more easily. Non 
+line-breaking whitespace is permitted, as is a subsequent comment. Hence 
+these examples are all syntactically valid, and are treated identically:
+
+```Dockerfile
+#DIRECTIVE=value
+# DIRECTIVE =value
+#	DIRECTIVE= value
+# DIRECTIVE = value
+#	  dIrEcTiVe=value
+```
+
+The following parser directive is supported:
+
+* `ESCAPE`
+
+## ESCAPE
```

Comment:
can you change this to a heading 3, so that it's listed *below* "directives" in the navigation?

```Markdown
### ESCAPE
```

Also, to prevent this being confused with an *instruction*, perhaps prefix the name with `#`, so:

```Markdown
### # ESCAPE
```

Would that make sense?